numpy/lib/function_base.py

```diff
@@ -547,12 +548,12 @@ def average(a, axis=None, weights=None, returned=False, *,
             raise ZeroDivisionError(
                 "Weights sum to zero, can't be normalized")
 
-        avg = np.multiply(a, wgt,
+        avg = avg_as_array = np.multiply(a, wgt,
```

Comment:
This looked a little weird at first, but I checked that the simpler diff doesn't cut it b/c of one test:

```diff
--- a/numpy/lib/function_base.py
+++ b/numpy/lib/function_base.py
@@ -516,6 +516,7 @@ def average(a, axis=None, weights=None, returned=False, *,
 
     if weights is None:
         avg = a.mean(axis, **keepdims_kw)
+        avg = np.asanyarray(avg)
         scl = avg.dtype.type(a.size/avg.size)
     else:
         wgt = np.asanyarray(weights)

```

```
==================================================================================================================================== FAILURES ====================================================================================================================================
______________________________________________________________________________________________________________________ TestMemmap.test_ufunc_return_ndarray ______________________________________________________________________________________________________________________
[gw2] linux -- Python 3.10.4 /home/treddy/python_virtual_envs/python_310_scipy_general/bin/python

self = <numpy.core.tests.test_memmap.TestMemmap object at 0x7f3d17af23e0>

    def test_ufunc_return_ndarray(self):
        fp = memmap(self.tmpfp, dtype=self.dtype, shape=self.shape)
        fp[:] = self.data
    
        with suppress_warnings() as sup:
            sup.filter(FutureWarning, "np.average currently does not preserve")
            for unary_op in [sum, average, product]:
                result = unary_op(fp)
>               assert_(isscalar(result))
E               AssertionError

fp         = memmap([[ 0.,  1.,  2.,  3.],
        [ 4.,  5.,  6.,  7.],
        [ 8.,  9., 10., 11.]], dtype=float32)
result     = array(5.5, dtype=float32)
self       = <numpy.core.tests.test_memmap.TestMemmap object at 0x7f3d17af23e0>
sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7f3d0ff3d600>
unary_op   = <function average at 0x7f3d1d18b2e0>

numpy/core/tests/test_memmap.py:158: AssertionError

```